### DISCUSSION TITLE
Rustdesk SELinux Log spam / sudo access

### DISCUSSION THREAD

**Discussion by bentalexanderhaase** [rustdesk/rustdesk#6302] [General]
### Bug Description

Rustdesk spams systems logs (about every 1s) with denied access to `sudo` due to SELinux.

```bash
~ # journalctl --boot 0 --follow
[...]
Nov 06 06:15:29 fedora setroubleshoot[23731]: SELinux is preventing rustdesk from execute access on the file /usr/bin/sudo. For complete SELinux messages run: sealert -l aad61d44-49b7-415b-95d3-aaa7dfa04e9f
Nov 06 06:15:29 fedora setroubleshoot[23731]: SELinux is preventing rustdesk from execute access on the file /usr/bin/sudo.
                                              
                                              *****  Plugin catchall (100. confidence) suggests   **************************
                                              
                                              If you believe that rustdesk should be allowed execute access on the sudo file by default.
                                              Then you should report this as a bug.
                                              You can generate a local policy module to allow this access.
                                              Do
                                              allow this access for now by executing:
                                              # ausearch -c 'rustdesk' --raw | audit2allow -M my-rustdesk
                                              # semodule -X 300 -i my-rustdesk.pp

```

### How to Reproduce

Install Rustdesk on Fedora 38 (soon 39), run Rustdesk.

Show logs using: `journalctl --boot 0 --follow`

### Expected Behavior

Rustdesk should not constantly try to gain `sudo` access when doing nothing (no connection in/outbound) unless user requested it to do so. Giving Rustdesk (client) total system access as [shown here](https://rustdesk.com/docs/en/client/linux/selinux/) via custom policy is not the correct solution. 

### Operating system(s) on local side and remote side

Fedora 38

### RustDesk Version(s) on local side and remote side

1.2.2

### Screenshots

See log above.

### Additional Context

Why exactly is Rustdesk trying to gain access to `sudo` for when doing nothing (Can someone point me to the code snippet)?

**Comment by sepkov** [comment]
Same on Fedora 39 arm64

**Comment by aanno** [comment]
It is not easy at all to get an RPM of rustdesk that really _works_ on Fedora.

* [discussion on making official RPMs](https://discussion.fedoraproject.org/t/can-we-have-rustdesk-in-fedora-default-repo/41293)
* [selinux problems with rustdesk](https://bugzilla.redhat.com/show_bug.cgi?id=2262995)

IMHO, it is very unlikely to get sane rustdesk RPMs working on selinux without major efforts.

**Reply by justintime4tea** [reply]
There is documentation provided by RustDesk that you can use to generate a "rustdesk-selinux" RPM package to accompany the official RustDesk RPM. See [my recent comment](https://github.com/rustdesk/rustdesk/discussions/6302#discussioncomment-9932512) for links and considerations.

**Comment by outsidefactor** [comment]
Has this stalled?

I am being impacted by this and there doesn't seem to have been any movement since Feb 9th?

EDIT:

For me, clean installs of 39 and 40 seem to work fine, but if you upgrade from 38 to 39 then Rustdesk never seems to work again. Is this the same for everyone else?

**Reply by justintime4tea** [reply]
The official RPM may have, at one point, included an SELinux policy. 

After following the guide from RustDesk official webisite, [link in my recent reply](https://github.com/rustdesk/rustdesk/discussions/6302#discussioncomment-9932512), RustDesk operates fine. 

I strongly suggest considering the security implications of the SELinux guide, however, and try to understand the overreaching power that is being given to RustDesk in the process of following the guide. It's best to make the most informed choice you can.

**Comment by justintime4tea** [comment]
I'm not sure if any of you have had the chance to take a look at the documentation. There is a section specifically related to SELinux:

- https://rustdesk.com/docs/en/client/linux/selinux/

If any of y'all have questions regarding SELinux in general, or crafting a policy package for RustDesk, I'd be glad to help! :) 

I'm working on creating a constrained SELinux policy package instead of using the one provided because I have security concerns related to the one provided (see below). It's low priority for me, however, given that I only use RustDesk client features from my own machine.

## ! WARNING !

The SELinux guide provided by RustDesk (above link), essentially, incorporates all of `init.te` into the `rustdesk.te` file and guides you through allowing every permission that shows as denied, for RustDesk, within the SELinux audit log. I'm certain RustDesk does not require _all_ the permissions of `init.te` nor should we simply add any/all permissions that show up as denied in the audit log.

**Be careful if you choose to follow** the [Enable through rpm installation](https://rustdesk.com/docs/en/client/linux/selinux/#enable-through-rpm-installation) section. The `rustdesk.te` file that is generated by `sepolicy generate` contains `permissive rustdesk_t` which essentially allows RustDesk to do whatever the heck it likes. 

You should remove the `permissive rustdesk_t` line if using `sepolicy generate` (RPM or make guide). _Personally_, I'd rather have a partially failing RustDesk than one that can do whatever it wants, and I'm none the wiser until I decide to dig through SELinux audit logs...

### Considerations if keeping "permissive rustdesk_t":
    
    With "permissive rustdesk_t", any "denials" are still logged. RustDesk is allowed to do whatever
    it wants but at least decisions from SELinux, like denying an action, are still logged just not 
    enforced. 

    The bad news though... you're essentially giving RustDesk "utlimate root", for lack of a better 
    description, and that feels very... very icky. 

    This is meant to allow you to craft an SELinux policy package, adding more rules through 
    audit2allow, but then you're expected to remove it afterward.
